exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java

```diff
@@ -1,169 +1,179 @@
-		System.out.println(players.get(currentPlayer)+ " was sent to the penalty box");
-		inPenaltyBox[currentPlayer] = true;
-		
-		currentPlayer++;
-		if (currentPlayer == players.size()) currentPlayer = 0;
-		return true;
-	}
-
-
-	private boolean didPlayerWin() {
-		return !(purses[currentPlayer] == 6);
-	}
-}
+package com.adaptionsoft.games.uglytrivia;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+
+public class Game {
+
+	private static final int MAX_PLAYERS = 6;
+	private static final int MIN_PLAYERS = 2;
+	private static final int WINNING_SCORE = 6;
+	private static final int MAX_QUESTIONS = 50;
+	private static final int LAST_POSITION = 11;
+	private static final int RESET_POSITION = 12;
+	private static final int PENALTY_CHECK_VALUE = 2;
+
+	private static final String CATEGORY_POP = "Pop";
+	private static final String CATEGORY_SCIENCE = "Science";
+	private static final String CATEGORY_SPORTS = "Sports";
+	private static final String CATEGORY_ROCK = "Rock";
+
+	private ArrayList<Player> playersInGame = new ArrayList();
+	private int[] playerOrder = new int[MAX_PLAYERS];
+	private int[] playerPurses = new int[MAX_PLAYERS];
+	private boolean[] playersInPenaltyBox = new boolean[MAX_PLAYERS];
+
+	private LinkedList popQuestionsList = new LinkedList();
+	private LinkedList scienceQuestionsList = new LinkedList();
+	private LinkedList sportsQuestionsList = new LinkedList();
+	private LinkedList rockQuestionsList = new LinkedList();
+
+	private int playerLocation = 0;
+	private boolean isOutOfPenaltyBox;
+
+	private Board gameBoard = new Board();
+
+    public Game(){
+    	for (int i = 0; i < MAX_QUESTIONS; i++) {
+			popQuestionsList.addLast("Pop Question " + i);
+			scienceQuestionsList.addLast(("Science Question " + i));
+			sportsQuestionsList.addLast(("Sports Question " + i));
+			rockQuestionsList.addLast("Rock Question " + i);
+    	}
+    }
+	
+	public boolean sufficientPlayersToPlay() {
+		return (getAmountOfPlayers() >= MIN_PLAYERS);
+	}
+
+	public boolean addPlayerToGame(Player player) {
+	    playersInGame.add(player);
+	    playerOrder[getAmountOfPlayers()] = 0;
+	    playerPurses[getAmountOfPlayers()] = 0;
+	    playersInPenaltyBox[getAmountOfPlayers()] = false;
+	    
+	    System.out.println(player + " was added");
+	    System.out.println("They are player number " + playersInGame.size());
+		return true;
+	}
+	
+	private int getAmountOfPlayers() {
+    	return playersInGame.size();
+	}
+
+	public void playGame(int roll) {
+    	Player player = playersInGame.get(playerLocation);
+		System.out.println(playersInGame.get(playerLocation) + " is the current player");
+		System.out.println("They have rolled a " + roll);
+		
+		if (player.isPlayerInPenaltyBox()) {
+			penaltyCheck(roll);
+		} else {
+			updatePlayerLocation(roll);
+		}
+	}
+
+	private void penaltyCheck(final int result) {
+		Player player = playersInGame.get(playerLocation);
+
+		if (result % PENALTY_CHECK_VALUE != 0) {
```

Comment:
Maybe this condition could be extracted into a method saying what it is checking eg 'isOddNumber'